### PR TITLE
[Transform] Make transform `_preview` request cancellable

### LINKED ISSUE #91286: [Transform] Make preview calls cancellable and respect timeout
A preview call can be quite expensive and cause timeout in a cluster. As a short term fix we identified 2 (hopefully) small fixes:

- make the call cancellable, the main issue here seems to be that preview calls search without using a parent task id(see this fix for data feeds: https://github.com/elastic/elasticsearch/pull/88142/files#diff-aba58244cb8f9dfa3ac3b6efeb02259e3d5956576f56c1ff5e243dd2aac27381)
- pass the timeout parameter of the preview call to the search call, currently a timeout can be specified, but is ignored for the inner search call, which is executed with no timeout (see https://www.elastic.co/guide/en/elasticsearch/reference/master/search-search.html#search-search-api-query-params).

Both changes are bugs and shall be relatively simple to implement and considered for backport to the 7.17 branch.

@przemekwitek can you have a look at this?

related: #88010

### PR DESCRIPTION
**Pull request by przemekwitek**
This PR improves the way `_preview` request is handled in Transform API:
- makes the `_preview` call cancellable by passing parent task id around
- makes the `_preview` time-bounded by passing `timeout` parameter around

Closes https://github.com/elastic/elasticsearch/issues/91286

### CODE REVIEW

File: x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/action/TransportPreviewTransformAction.java [hunk 1]
```diff
@@ -112,6 +115,7 @@ public TransportPreviewTransformAction(
 
     @Override
     protected void doExecute(Task task, Request request, ActionListener<Response> listener) {
+        TaskId parentTaskId = new TaskId(clusterService.getNodeName(), task.getId());
```
Comment (droberts195) [inline x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/action/TransportPreviewTransformAction.java:118]: This should use the node ID, not the node name.

(I guess you copied this from one of the ML PRs. If you look at the current code you’ll see there was a followup to fix this mistake.)

Comment (przemekwitek) [reply x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/action/TransportPreviewTransformAction.java:118]: Ooops, that's true.

Done.

**Review by hendrikmuhs — APPROVED** [review]
LGTM

**Review by droberts195 — APPROVED** [review]
LGTM2

**Comment by przemekwitek** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|7.17|[](https://github.com/elastic/elasticsearch/pull/91389)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**